### PR TITLE
feat(authentication-child): SignUp child flow

### PR DESCRIPTION
**Pull request by akhlivnoy**
* add new action saveSignUpDataParent
* add new action saveSignUpDataChild
* modify SignUp navigation
* add SignUp child flow

### CODE REVIEW

File: app/redux/index.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ sagaMiddleware.run(rootSaga);
 export const persistor = persistStore(store);
 
 // Unсomment this line to clear the cache
-// persistor.purge();
+//persistor.purge();
```
Comment (emberglaz) [inline app/redux/index.ts]: return the space

**Review by emberglaz — CHANGES_REQUESTED** [review]

File: app/navigation/stacks/jointOnboardingStackNavigator.tsx [hunk 2]
```diff
@@ -21,5 +25,7 @@ export const JointOnboardingStackNavigator = () => (
     <Stack.Screen name="Loading" component={LoadingScreen} />
     <Stack.Screen name="WelcomeChild" component={WelcomeChildScreen} />
     <Stack.Screen name="SidekickIntro" component={SidekickIntroScreen} />
+    <Stack.Screen name="SignUpNickname" component={SignUpNicknameScreen} />
+    <Stack.Screen name="SignUpAvatar" component={SignUpAvatarScreen} />
```
Comment (Reykjavik151) [inline app/navigation/stacks/jointOnboardingStackNavigator.tsx]: https://stackoverflow.com/questions/60115818/initial-route-params-in-react-navigation-v5
use predefined initial params

File: app/navigation/stacks/parentsOnboardingStackNavigator.tsx [hunk 3]
```diff
@@ -18,8 +18,8 @@ export type ParentsOnboardingStackParams = {
   LanguageSelection: undefined;
   Acknowledgement: undefined;
   SignUpCredentials: undefined;
-  SignUpNickname: undefined;
-  SignUpAvatar: undefined;
+  SignUpNickname: { isChild: boolean };
+  SignUpAvatar: { isChild: boolean };
   CharmsIntroducing: undefined;
   OnboardingCarousel: undefined;
 };
```
Comment (Reykjavik151) [inline app/navigation/stacks/parentsOnboardingStackNavigator.tsx]: the same as above -- use predefined route params
https://stackoverflow.com/questions/60115818/initial-route-params-in-react-navigation-v5

File: app/screens/Onboarding/SignUp/SignUpCredentials/SignUpCredentials.tsx [hunk 4]
```diff
@@ -23,8 +23,8 @@ export const SignUpCredentialsScreen: React.FC<ISignUpCredentialsScreenProps> =
     const [password, setPassword] = useState('');
 
     const onSignUpPress = useCallback(() => {
-      dispatch(cacheSlice.actions.saveSignUpData({ email, password }));
-      navigation.navigate('SignUpNickname');
+      dispatch(cacheSlice.actions.saveSignUpDataParent({ email, password }));
+      navigation.navigate('SignUpNickname', { isChild: false });
```
Comment (Reykjavik151) [inline app/screens/Onboarding/SignUp/SignUpCredentials/SignUpCredentials.tsx]: so if you do the changes described above -- you won't pass it here

File: app/screens/Onboarding/SignUp/SignUpNickname/SignUpNickname.tsx [hunk 5]
```diff
@@ -7,25 +7,31 @@ import {
...
   const onSubmit = useCallback(() => {
-    dispatch(cacheSlice.actions.saveSignUpData({ nickname }));
-    navigation.navigate('SignUpAvatar');
-  }, [dispatch, navigation, nickname]);
+    if (isChild) {
+      dispatch(cacheSlice.actions.saveSignUpDataChild({ nickname }));
+    } else {
+      dispatch(cacheSlice.actions.saveSignUpDataParent({ nickname }));
+    }
+    navigation.navigate('SignUpAvatar', { isChild });
```
Comment (Reykjavik151) [inline app/screens/Onboarding/SignUp/SignUpNickname/SignUpNickname.tsx]: the same as above

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

**Review by Reykjavik151 — APPROVED** [review]

**Pull request merged**